File: Identity/src/main/java/com/badasstechie/identity/controller/AuthController.java
package com.badasstechie.identity.controller;

import com.badasstechie.identity.dto.AuthRequest;
import com.badasstechie.identity.dto.AuthResponse;
import com.badasstechie.identity.dto.RefreshTokenRequest;
import com.badasstechie.identity.service.AuthService;
import com.badasstechie.identity.service.RefreshTokenService;
import io.swagger.v3.oas.annotations.Operation;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/v1/identity/auth")
@RequiredArgsConstructor
public class AuthController {
    private final AuthService authService;
    private final RefreshTokenService refreshTokenService;

    @PostMapping("/auth_token")
    @Operation(summary = "Request an access token")
    public AuthResponse requestAccessToken(@RequestBody AuthRequest authRequest) {
        return authService.requestAccessToken(authRequest);
    }

    @PostMapping("/refresh_token")
    @Operation(summary = "Refresh an access token")
    public AuthResponse refreshToken(@RequestBody RefreshTokenRequest refreshTokenRequest) {
        return authService.refreshAccessToken(refreshTokenRequest);
    }

    @GetMapping("/logout/{email}")
    @Operation(summary = "Clear all your refresh tokens")
    public ResponseEntity<String> logout(@PathVariable String email) {
        refreshTokenService.deleteRefreshTokensByUser(email);
        return new ResponseEntity<>("You will be signed out shortly", HttpStatus.ACCEPTED);
    }
}

File: Identity/Controllers/AuthController.cs
using Identity.DTOs;
using Identity.Services;
using Microsoft.AspNetCore.Mvc;

namespace Identity.Controllers;

[ApiController]
[Route("api/v1/identity/auth")]
public class AuthController : ControllerBase
{
    private readonly IAuthService _authService;
    private readonly IRefreshTokenService _refreshTokenService;

    public AuthController(IAuthService authService, IRefreshTokenService refreshTokenService)
    {
        _authService = authService;
        _refreshTokenService = refreshTokenService;
    }

    /// <summary>
    /// Request an access token
    /// </summary>
    [HttpPost("auth_token")]
    public async Task<ActionResult<AuthResponse>> RequestAccessToken([FromBody] AuthRequest request)
    {
        try
        {
            var response = await _authService.RequestAccessTokenAsync(request);
            return Ok(response);
        }
        catch (UnauthorizedAccessException ex)
        {
            return Unauthorized(new { message = ex.Message });
        }
    }

    /// <summary>
    /// Refresh an access token
    /// </summary>
    [HttpPost("refresh_token")]
    public async Task<ActionResult<AuthResponse>> RefreshToken([FromBody] RefreshTokenRequest request)
    {
        try
        {
            var response = await _authService.RefreshAccessTokenAsync(request);
            return Ok(response);
        }
        catch (UnauthorizedAccessException ex)
        {
            return Unauthorized(new { message = ex.Message });
        }
    }

    /// <summary>
    /// Clear all your refresh tokens
    /// </summary>
    [HttpGet("logout/{email}")]
    public async Task<ActionResult<string>> Logout(string email)
    {
        await _refreshTokenService.DeleteRefreshTokensByUserAsync(email);
        return Accepted("You will be signed out shortly");
    }
}
